src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -1525,67 +1525,12 @@ splitSuffixWith :: Monad m
 splitSuffixWith predicate f = foldMany1 (FL.sliceSepWith predicate f)
 
 {-# INLINE_NORMAL groupsBy #-}
-groupsBy :: Monad m
+groupsBy :: MonadThrow m
     => (a -> a -> Bool)
     -> Fold m a b
     -> Stream m a
     -> Stream m b
-groupsBy cmp f (Stream step state) = Stream (stepOuter f) (Just state, Nothing)
-
-    where
-
-    {-# INLINE_LATE stepOuter #-}
-    stepOuter (Fold fstep initial done) gst (Just st, Nothing) = do
-        res <- step (adaptState gst) st
-        case res of
-            Yield x s -> do
-                fs <- initial
-                sfs <- fstep fs x
-                case sfs of
-                    FL.Partial fs1 -> go SPEC x s fs1
-                    FL.Done fb -> return $ Yield fb (Just s, Just x)
-            Skip s -> return $ Skip $ (Just s, Nothing)
-            Stop -> return Stop
-
-        where
-
-        go !_ prev stt !acc = do
-            res <- step (adaptState gst) stt
-            case res of
-                Yield x s -> do
-                    if cmp x prev
-                    then do
-                        sfs <- fstep acc x
-                        case sfs of
-                            FL.Partial fs1 -> go SPEC prev s fs1
-                            FL.Done fb -> return $ Yield fb (Just s, Just x)
-                    else done acc >>= \r -> return $ Yield r (Just s, Just x)
-                Skip s -> go SPEC prev s acc
-                Stop -> done acc >>= \r -> return $ Yield r (Nothing, Nothing)
-    stepOuter (Fold fstep initial done) gst (Just st, Just prev) = do
-        fs <- initial
-        sfs <- fstep fs prev
-        case sfs of
-            FL.Partial fs1 -> go SPEC st fs1
-            FL.Done fb -> return $ Yield fb (Just st, Nothing)
-
-        where
-
-        -- XXX code duplicated from the previous equation
-        go !_ stt !acc = do
-            res <- step (adaptState gst) stt
-            case res of
-                Yield x s -> do
-                    if cmp x prev
-                    then do
-                        sfs <- fstep acc x
-                        case sfs of
-                            FL.Partial fs1 -> go SPEC s fs1
-                            FL.Done fb -> return $ Yield fb (Just s, Just x)
-                    else done acc >>= \r -> return $ Yield r (Just s, Just x)
-                Skip s -> go SPEC s acc
-                Stop -> done acc >>= \r -> return $ Yield r (Nothing, Nothing)
-    stepOuter _ _ (Nothing, _) = return Stop
+groupsBy cmp f = parseMany (PRD.groupBy (\a b -> cmp b a) f)
```

Comment:
We can keep the old definition for now to avoid the MonadCatch constraint. When parsers are changed to remove MonadCatch we can make this change. You can add this definition commented out and also add this idiom in haddock of `groupsBy` in Prelude.

Reply:
I'm not adding the documentation in Prelude. We should deprecate it in prelude according to #563

Reply:
> You can add this definition commented out and also add this idiom in haddock of groupsBy in Prelude.

Will do it in the PR, deprecating `groupsBy`. Adding `... PRD.groupBy (flip cmp) ...` is weird.